pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -326,11 +327,54 @@ func (c *Cluster) postReconcileUpdateOSDProperties(desiredOSDs map[int]*OSDInfo)
 	}
 	logger.Debugf("post processing osd properties with %d actual osds from ceph osd df and %d existing osds found during reconcile", len(osdUsage.OSDNodes), len(desiredOSDs))
 	for _, actualOSD := range osdUsage.OSDNodes {
-		if desiredOSD, ok := desiredOSDs[actualOSD.ID]; ok {
-			if err := c.updateDeviceClassIfChanged(actualOSD.ID, desiredOSD.DeviceClass, actualOSD.DeviceClass); err != nil {
-				// Log the error and allow other updates to continue
-				logger.Error(err)
-			}
+		if err := c.resizeOsdCrushWeight(actualOSD); err != nil {
+			// Log the error and allow other updates to continue
+			logger.Error("failed to resize osd crush weight on cluster in namespace %s %v", c.clusterInfo.Namespace, err)
+		}
+		desiredOSD, ok := desiredOSDs[actualOSD.ID]
+		if !ok {
+			continue
+		}
+		if err := c.updateDeviceClassIfChanged(actualOSD.ID, desiredOSD.DeviceClass, actualOSD.DeviceClass); err != nil {
+			// Log the error and allow other updates to continue
+			logger.Error("failed to update device class on cluster in namespace %s %v", c.clusterInfo.Namespace, err)
+		}
+	}
+	return nil
+}
+
+func convertKibbytesToTibbytes(kib string) (float64, error) {
+	kibFloat, err := strconv.ParseFloat(kib, 64)
+	if err != nil {
+		return float64(0), fmt.Errorf("failed converting string to float: %v", err)
```

Comment:
```suggestion
		return float64(0), errors.Wrap(err, "failed to convert string to float")
```

Reply:
In general how should we decide between  fmt.Errorf vs errors.Wrap?

Reply:
If we're returning an error, our typical pattern is to use Wrap.